homeassistant/components/tag/__init__.py

```diff
@@ -132,8 +203,91 @@ async def async_scan_tag(
         context=context,
     )
 
+    _add_device_id = {}
+    if device_id:
+        _add_device_id[DEVICE_ID] = device_id
     if tag_id in helper.data:
-        await helper.async_update_item(tag_id, {LAST_SCANNED: dt_util.utcnow()})
+        await helper.async_update_item(
+            tag_id, {LAST_SCANNED: dt_util.utcnow(), **_add_device_id}
+        )
     else:
-        await helper.async_create_item({TAG_ID: tag_id, LAST_SCANNED: dt_util.utcnow()})
+        await helper.async_create_item(
+            {TAG_ID: tag_id, LAST_SCANNED: dt_util.utcnow(), **_add_device_id}
+        )
     _LOGGER.debug("Tag: %s scanned by device: %s", tag_id, device_id)
+
+
+class TagEntity(Entity):
+    """Representation of a Tag entity."""
+
+    _unrecorded_attributes = frozenset({TAG_ID})
+    _attr_translation_key = DOMAIN
+    _attr_should_poll = False
+
+    # Implements it's own platform
+    _no_platform_reported = True
```

Comment:
Why is this needed? If we get warnings about the entity being created without a platform, something is wrong and we should now hide it IMHO.